octobot_services/services/service_factory.py

```diff
@@ -0,0 +1,55 @@
+#  Drakkar-Software OctoBot-Services
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+from octobot_commons.constants import CONFIG_SERVICE_FILE
+
+from octobot_commons.logging.logging_util import get_logger
+
+from octobot_services.constants import CONFIG_CATEGORY_SERVICES, CONFIG_SERVICE_INSTANCE
+from octobot_services.services.abstract_service import AbstractService
+
+
+class ServiceFactory:
+    def __init__(self, config, backtesting_enabled):
+        self.logger = get_logger(self.__class__.__name__)
+        self.config = config
+        self.backtesting_enabled = backtesting_enabled
+
+    async def create_all(self, services_tentacles_path=CONFIG_SERVICE_FILE):
+        for service_class in AbstractService.__subclasses__():
+            await self._create_service(service_class, services_tentacles_path)
+
+    # could be public, left as protected since not used so far
+    async def _create_service(self, service_class, services_tentacles_path):
+        service_instance = service_class()
+        if service_instance.get_is_enabled(self.config) and \
+                (not self.backtesting_enabled or service_instance.BACKTESTING_ENABLED):
```

Comment:
I think we should manage service creation when backtesting in https://github.com/Drakkar-Software/OctoBot, not here.